tests/basic/storage_pool_test.go

```diff
@@ -9761,3 +9761,192 @@ var _ = Describe("{AddDriveMetadataPool}", func() {
+			if selectedPool != nil {
+				drvMap, err := Inst().V.GetPoolDrives(&selectedNode)
+				if len(drvMap) == 0 {
+					log.FailOnError(fmt.Errorf("error getting the drive map from the node"), "error getting pool drivemap from node [%s]", selectedNode.Name)
+				}
+				log.FailOnError(err, "error getting pool drives from node [%s]", selectedNode.Name)
+				drvs := drvMap[fmt.Sprintf("%d", selectedPool.ID)]
+				maxDrivesAllowed = POOL_MAX_CLOUD_DRIVES - len(drvs)
+			} else {
+				log.FailOnError(err, "error getting pool from node [%s]", selectedNode.Name)
+			}
+			for i := 1; i <= maxDrivesAllowed+1; i++ {
+				drvSize, err := getPoolDiskSize(selectedPool)
+				log.FailOnError(err, "fialed while getting pool size")
+				driveSize := drvSize * uint64(i)
+				expectedSize := (selectedPool.TotalSize / units.GiB) + driveSize
+				err = Inst().V.ExpandPool(selectedPool.Uuid, api.SdkStoragePool_RESIZE_TYPE_ADD_DISK, expectedSize, false)
+				log.FailOnError(err, "error while expanding pool")
+				isjournal, err := isJournalEnabled()
+				log.FailOnError(err, "Failed to check is journal enabled")
+				resizeErr := waitForPoolToBeResized(expectedSize, selectedPool.Uuid, isjournal)
+				if i == maxDrivesAllowed+1 {
+					poolStatus, err := getPoolLastOperation(selectedPool.Uuid)
+					log.FailOnError(err, "error getting pool status")
+					dash.VerifyFatal(resizeErr != nil, true, fmt.Sprintf("Expected new size to be '%d' or '%d'", expectedSize, expectedSize-3))
+					if poolStatus != nil {
+						dash.VerifyFatal(strings.Contains(poolStatus.Msg, "unable to add 1 new drive(s) as it would exceed maximum supported drives (6)"), true, "Error expected as drive added more than allowed per pool")
+					} else {
+						err := errors.New("failed while getting msg")
+						log.FailOnError(err, "failed while getting msg")
+					}
+				} else {
+					dash.VerifyFatal(resizeErr != nil, false, fmt.Sprintf("Expected new size to be '%d' or '%d'", expectedSize, expectedSize-3))
+				}
+			}
+			err = Inst().V.RefreshDriverEndpoints()
+			log.FailOnError(err, "error refreshing volume endpoints")
+		})
+
+		Step("Test beyond max pool count", func() {
+			log.InfoD(stepLog)
+			poolList, err := GetPoolsDetailsOnNode(selectedNode)
+			poolLeft := maxPoolLength - len(poolList)
+			err = addNewPools(selectedNode, poolLeft)
+			log.FailOnError(err, "error adding new pools")
+			err = addNewPools(selectedNode, 1)
+			dash.VerifyFatal(err != nil, true, "Error expected as pools trying to create is more than allowed")
+			poolList, err = GetPoolsDetailsOnNode(selectedNode)
+		})
+
+		Step("Test beyond max drives per node", func() {
+			log.InfoD(stepLog)
+			var selectedPools string
+			poolList, err := GetPoolsDetailsOnNode(selectedNode)
+			log.FailOnError(err, "failed while getting updated node pool list")
+			//first add all the new pools
+			poolLeft := maxPoolLength - len(poolList)
+			err = addNewPools(selectedNode, poolLeft)
```

Comment:
how are you validating that drive limit is not reached while adding new pools?

Reply:
poolleft will give the max pools that can be added

Reply:
if pools left 4 bit max drive allowed is only 2, how is this handled?